### ISSUE TITLE
#[async_trait] with 'async_trait lifetime is incorrectly flagged as being an experimental in-band lifetime

### ISSUE THREAD

**Issue by Silvea12** [intellij-rust/intellij-rust#7863]
## Environment

* **IntelliJ Rust plugin version:** 0.4.155.4114-212
* **Rust toolchain version:** 1.55.0 (c8dfcfe04 2021-09-06) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2021.2.2 (CL-212.5284.51)
* **Operating system:** Linux 5.10.63-1-manjaro
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description

When using `#[async_trait]`, which declares a lifetime internally for 'async_lifetime, the Rust plugin incorrectly flags it as being [E0658 - "in-band lifetimes is experimental"](https://doc.rust-lang.org/error-index.html#E0658)

## Steps to reproduce

```rs
use async_trait::async_trait;

#[async_trait]
trait MyTrait {
    async fn my_fn<T: AsRef<()> + Send + Sync + 'async_trait>();
}
```

`'async_trait` will be flagged as being an experimental in-band lifetime, despite macro expansion expanding to

```rs
trait MyTrait {
    #[must_use]
    #[allow(clippy::type_complexity, clippy::type_repetition_in_bounds)]
    fn my_fn<'async_trait, T: AsRef<()> + Send + Sync + 'async_trait>() -> ::core::pin::Pin<
        Box<dyn ::core::future::Future<Output = ()> + ::core::marker::Send + 'async_trait>,
    >
        where
            T: 'async_trait;
}
```

**Comment by Silvea12** [comment]
As of 0.4.166.4486-213, this functionality has been broken.

I've also seen a regression where a wildcard use fails too for async_trait - the plugin fails to resolve the member fn unless it's directly included.

<details>
<summary>Minimal working/compilable sample:</summary>

main.rs:
```rs
mod vec_ext;

use vec_ext::*;
// Swapping to the below one makes intellij-rust happy
// use vec_ext::VecExt;

fn main() {
    let v = vec![1,2,3];
    v.print();
}
```

vec_ext.rs:
```rs
use async_trait::async_trait;

#[async_trait]
pub trait VecExt {
    fn print(&self);
}

#[async_trait]
impl<T: std::fmt::Debug> VecExt for std::vec::Vec<T> {
    fn print(&self) {
        println!("{self:?}");
    }
}
```
</details>

**Comment by webern** [comment]
I find that `async::trait` makes jump-to-definition, code completion, etc unusable. I'm looking for a tracking issue specific to these challenges with `async::trait`. Is there an issue tracking general breakdown of the code analyzer when it encounters `async::trait`?
